compiler/src/dotty/tools/dotc/core/TypeApplications.scala

```diff
@@ -533,6 +533,9 @@ class TypeApplications(val self: Type) extends AnyVal {
     case JavaArrayType(elemtp) => elemtp
     case tp: OrType if tp.tp1.isBottomType => tp.tp2.elemType
     case tp: OrType if tp.tp2.isBottomType => tp.tp1.elemType
-    case _ => self.baseType(defn.SeqClass).argInfos.headOption.getOrElse(NoType)
+    case _ =>
+      self.baseType(defn.SeqClass)
+      .orElse(self.baseType(defn.ArrayClass))
```

Comment:
How come the ArrayOf case above isn't enough? Should the ArrayOf extractor be generalized instead?

Reply:
Hmm maybe not since `Array[Int] | Array[String] <:< Array[? >: Int & String <: Int | String]` but not vice-versa, so we'd loose the property that `case ArrayOf(x) => ArrayOf(x)` return the same type, which might be confusing.

Reply:
I tried `tp.baseType(ArrayClass).argInfos.headOption` in `ArrayOf` and 30+ run tests failed (before I killed the local run).  And those were backend failures.